_posts/2024-01-29-good-parallel-computer.md

```diff
@@ -0,0 +1,74 @@
+---
+layout: post
+title:  "I want a good parallel computer"
+date:   2024-01-29 10:30:42 -0800
+categories: [gpu]
+---
+Much of my research over the past few years has been 2D vector graphics rendering on GPUs. That work goes well, but I am running into the limitations of GPU hardware and programming interfaces, and am starting to see hints that a much better parallel computer may be possible. At the same time, I see some challenges regarding actually getting there. This essay will explore both in depth.
+
+I should qualify, the workload I care about is unusual in a number of respects. Most game workloads involve rasterization of a huge number of triangles, and most AI workloads involve multiplication of large matrices, both very conceptually simple operations. By contrast, 2D rendering has a lot of intricate, conditional logic, and is very compute intensive compared with the raw memory bandwidth needed. Compute shaders on modern GPUs can handle the conditional logic quite well, but lack *agility,* which to me means the ability to make fine-grained scheduling decisions. I believe agility 
+
```

Comment:
FYI unfinished sentence.

YYY gpu vs agility. Does this kind of code also hamper parallelism?